### PR TITLE
Remove use of distutils

### PR DESCRIPTION
**Pull request by edgarrmondragon**
* Closes https://github.com/AzureAD/microsoft-authentication-extensions-for-python/issues/121
* Closes https://github.com/AzureAD/microsoft-authentication-extensions-for-python/issues/118

### CODE REVIEW

**Comment by edgarrmondragon** [comment]
@microsoft-github-policy-service agree

File: msal_extensions/cache_lock.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def __init__(self, lockfile_path):
         self._lockpath = lockfile_path
         # Support for passing through arguments to the open syscall was added in v1.4.0
         open_kwargs = ({'buffering': 0}
-            if LooseVersion(portalocker.__version__) >= LooseVersion("1.4.0") else {})
+            if Version(portalocker.__version__) >= Version("1.4.0") else {})
```
Comment (edgarrmondragon) [inline msal_extensions/cache_lock.py:24]: I don't think `LooseVersion` is really needed here, so `Version` should be good enough.

Comment (rayluo) [reply msal_extensions/cache_lock.py:24]: The `LooseVersion()` was introduced by @bluca . What do you think with this change, Luca?

Comment (bluca) [reply msal_extensions/cache_lock.py:24]: looks good to me, thanks

**Review by bluca — APPROVED** [review]

**Comment by edgarrmondragon** [comment]
Thanks @bluca! Tests are passing for me locally, even on 2.7:

<details>
<summary>Test results on 2.7</summary>

```
$ pytest -vvv
============================================================================================== test session starts ===============================================================================================
platform darwin -- Python 2.7.18, pytest-4.6.11, py-1.11.0, pluggy-0.13.1 -- /oss/microsoft-authentication-extensions-for-python/venv2.7/bin/python
cachedir: .pytest_cache
rootdir: /oss/contrib/microsoft-authentication-extensions-for-python
collected 18 items / 1 skipped / 17 selected                                                                                                                                                                     

tests/test_agnostic_backend.py::test_file_token_cache_roundtrip SKIPPED                                                                                                                                    [  5%]
tests/test_agnostic_backend.py::test_current_platform_cache_roundtrip_with_persistence_builder SKIPPED                                                                                                     [ 11%]
tests/test_agnostic_backend.py::test_persisted_token_cache SKIPPED                                                                                                                                         [ 16%]
tests/test_agnostic_backend.py::test_file_not_found_error_is_not_raised PASSED                                                                                                                             [ 22%]
tests/test_cache_lock_file_perf.py::test_lock_for_normal_workload PASSED                                                                                                                                   [ 27%]
tests/test_cache_lock_file_perf.py::test_lock_for_high_workload PASSED                                                                                                                                     [ 33%]
tests/test_cache_lock_file_perf.py::test_lock_for_timeout PASSED                                                                                                                                           [ 38%]
tests/test_crossplatlock.py::test_ensure_file_deleted PASSED                                                                                                                                               [ 44%]
tests/test_macos_backend.py::test_keychain_roundtrip PASSED                                                                                                                                                [ 50%]
tests/test_macos_backend.py::test_osx_token_cache_roundtrip SKIPPED                                                                                                                                        [ 55%]
tests/test_persistence.py::test_file_persistence PASSED                                                                                                                                                    [ 61%]
tests/test_persistence.py::test_nonexistent_file_persistence PASSED                                                                                                                                        [ 66%]
tests/test_persistence.py::test_file_persistence_with_data_protection SKIPPED                                                                                                                              [ 72%]
tests/test_persistence.py::test_nonexistent_file_persistence_with_data_protection SKIPPED                                                                                                                  [ 77%]
tests/test_persistence.py::test_keychain_persistence PASSED                                                                                                                                                [ 83%]
tests/test_persistence.py::test_nonexistent_keychain_persistence PASSED                                                                                                                                    [ 88%]
tests/test_persistence.py::test_libsecret_persistence SKIPPED                                                                                                                                              [ 94%]
tests/test_persistence.py::test_nonexistent_libsecret_persistence SKIPPED                                                                                                                                  [100%]

===================================================================================== 10 passed, 9 skipped in 11.51 seconds ======================================================================================
```

</details>

So I pushed another commit to use `fail-fast: false` and let the non-2.7 tests run.

**Review by rayluo — APPROVED** [review]

**Comment by edgarrmondragon** [comment]
Hey folks 👋

Is it worth making a new release with this change?

**Pull request merged**